react/ProductImage.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { Fragment, FunctionComponent, useState } from 'react'
+import ReactResizeDetector from 'react-resize-detector'
+
+import { utils } from 'vtex.my-account-commons'
+
+const { fixImageUrl } = utils
+
+interface Props {
+  className?: string
+  url: string
+  alt: string
+}
+
+const ProductImage: FunctionComponent<Props> = ({
+  className,
+  url,
+  alt,
+}: Props) => {
+  const [maxWidth, setMaxWidth] = useState(50)
+  const [maxHeight, setMaxHeight] = useState(50)
+
+  const resizeHandler = (width: number, height: number) => {
+    if (width > maxWidth) {
+      setMaxWidth(width + 50)
+    }
+    if (height > maxHeight) {
+      setMaxHeight(height + 50)
+    }
+  }
+
+  return (
+    <Fragment>
+      <ReactResizeDetector
+        handleWidth
+        handleHeight
+        onResize={(width: number, height: number) =>
+          resizeHandler(width, height)
+        }
+      />
+      <img
+        className={`${className}`}
```

Comment:
```suggestion
        className={className}
```

Reply:
I would rather do:

```js
  <img src={fixImageUrl(url, maxWidth, maxHeight)} ...imageProps />
```

That way the user could set other attributes for the image tag, like width and height, etc.

Reply:
Updated to support imageProps, defined as: 

```javascript
imageProps?: {
    width?: number;
    height?: number;
    className?: string;
  }
```